sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/ScriptTransformation.scala

```diff
@@ -32,7 +32,13 @@ case class ScriptTransformation(
     child: LogicalPlan,
     ioschema: ScriptInputOutputSchema) extends UnaryNode {
   @transient
-  override lazy val references: AttributeSet = AttributeSet(child.output)
```

Comment:
This is not the only place that uses this pattern. Do you know which unresolved plan throw errors when calling `output`?

Reply:
Hi, we encountered the same problem in our production env. Is there any update on this?

Reply:
@cloud-fan     When the `output` of `'Project[UnresolvedStar]` is called, an error is thrown.

The SQL plan in the pr description is like this:
```tex
'ScriptTransformation cat, [a#6], ScriptInputOutputSchema(List(),List(),Some(org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe),Some(org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe),List((field.delim,	), (serialization.last.column.takes.rest,true)),List((field.delim,	), (serialization.last.column.takes.rest,true)),Some(org.apache.hadoop.hive.ql.exec.TextRecordReader),Some(org.apache.hadoop.hive.ql.exec.TextRecordWriter),false)
+- 'Project [*]
   +- 'SubqueryAlias __auto_generated_subquery_name
      +- 'Distinct
         +- 'Union false, false
            :- Project [c2#8 AS c#4]
            :  +- SubqueryAlias t2
            :     +- View (`t2`, [c1#7,c2#8])
            :        +- Project [cast(c1#9 as decimal(2,1)) AS c1#7, cast(c2#10 as int) AS c2#8]
            :           +- SubqueryAlias tbl
            :              +- LocalRelation [c1#9, c2#10]
            +- Project [c2#12 AS c#5]
               +- SubqueryAlias t1
                  +- View (`t1`, [c1#11,c2#12])
                     +- Project [cast(c1#13 as int) AS c1#11, cast(c2#14 as string) AS c2#12]
                        +- SubqueryAlias tbl
                           +- LocalRelation [c1#13, c2#14]
```